workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/utilities/TokenBuilder.kt

```diff
@@ -0,0 +1,84 @@
+package com.r3.corda.lib.tokens.workflows.utilities
+
+import com.r3.corda.lib.tokens.contracts.states.FungibleToken
+import com.r3.corda.lib.tokens.contracts.types.IssuedTokenType
+import com.r3.corda.lib.tokens.contracts.types.TokenType
+import com.r3.corda.lib.tokens.contracts.utilities.heldBy
+import com.r3.corda.lib.tokens.contracts.utilities.issuedBy
+import com.r3.corda.lib.tokens.contracts.utilities.of
+import net.corda.core.contracts.Amount
+import net.corda.core.identity.Party
+import java.lang.IllegalArgumentException
+import java.math.BigDecimal
+
+/**
+ * This is a utility class designed for Java developers to more easily
+ * access Kotlin infix functions to build token types.
+ */
+class TokenBuilder {
+    private var intAmount: Int? = null
+    private var longAmount: Long? = null
+    private var doubleAmount: Double? = null
+    private var bigDecimalAmount: BigDecimal? = null
+    private lateinit var amountTokenType: Amount<TokenType>
+    private lateinit var amountIssuedTokenType: Amount<IssuedTokenType>
+    private lateinit var fungibleToken: FungibleToken
+
+    private fun getInitializedAmount() = if (!amountInitialized()) {
+        listOfNotNull(intAmount, longAmount, doubleAmount, bigDecimalAmount).single()
+    } else {
+        throw IllegalArgumentException("The token amount hasn't been initialized")
+    }
+
+    private fun amountInitialized() = listOf(intAmount, longAmount, doubleAmount, bigDecimalAmount).all { it != null }
+    private fun initializeAmountOrThrow(callbackSetter: () -> Unit): TokenBuilder {
+        if (!amountInitialized()) {
+            callbackSetter()
+        } else {
+            throw IllegalArgumentException("The token amount has already been initialized")
+        }
+        return this
+    }
+
+    fun amount(intAmount: Int) = initializeAmountOrThrow { this.intAmount = intAmount }
+    fun amount(longAmount: Long) = initializeAmountOrThrow { this.longAmount = longAmount}
+    fun amount(doubleAmount: Double) = initializeAmountOrThrow { this.doubleAmount = doubleAmount }
+    fun amount(bigDecimalAmount: BigDecimal) = initializeAmountOrThrow { this.bigDecimalAmount = bigDecimalAmount }
+
+    fun <T: TokenType> of(t: T): TokenBuilder = when(getInitializedAmount()) {
```

Comment:
What is this method for?

Reply:
This is a replication of the `of` method - which is an infix function applied to a number value (`Int`, `Long`, `BigDecimal`, `Double`) and accepting either a `TokenType` or a subtype of `TokenType` (e.g. - issued `IssuedTokenType`) hence the generics.